options/rtdl/generic/main.cpp

```diff
@@ -485,6 +486,29 @@ int __dlapi_reverse(const void *ptr, __dlapi_symbol *info) {
 	return -1;
 }
 
+extern "C" [[ gnu::visibility("default") ]]
+int __dlapi_iterate_phdr(int (*callback)(struct dl_phdr_info *, size_t, void*), void *data) {
+	int last_return = 0;
+	for (auto object : globalScope->_objects) {
+		struct dl_phdr_info info;
+		info.dlpi_addr = object->baseAddress;
+		info.dlpi_name = object->name;
+		info.dlpi_phdr = static_cast<ElfW(Phdr)*>(object->phdrPointer);
+		info.dlpi_phnum = object->phdrCount;
+		info.dlpi_adds = rtsCounter;
+		info.dlpi_subs = 0; // TODO(geert): implement dlclose().
```

Comment:
`rtsCounter` is not exactly what we want here. As far as I can see, `dlpi_adds` is a global counter that is incremented whenever an object is loaded (?).

Reply:
Yes, that is ``rtsCounter`` though, right? Or is it also decremented at some point?

Reply:
Ah, true, I made a mistake there.